### ISSUE TITLE
Web fonts issues

### ISSUE THREAD

**Issue by vic-topcoder** [appirio-tech/topcoder-app#1117]
There seems to be a dependency issues with the web fonts stack on connect app. I traced declaration for Typekit, which we decided to not use. The declarations of Roboto font family is done in `_fonts.scss`, which is in `react-components`, but that is dependency from `connect-app`. We need to fix the use and declarations of web fonts for all apps and components.

1. Remove dependency of font declarations on `connect-app`;
2. Allow font declaration to be done with a similar to `_fonts.scss` way;
3. We want to modify the font stack to include only ".woff" and ".woff2" fonts, as we're not supporting old browser stack.

Webpack should load the fonts correctly on local environment and the AWS → Cloudfront environments. We may need to define
- CORS issues for CDN;
- MIME type for fonts;

An article that may help: https://shellmonger.com/2016/01/22/working-with-fonts-with-webpack/

We want to use our CDN to serve fonts; Google fonts and Adobe Typekit are not on the list of options right now.
http://caniuse.com/#search=web%20font;
https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face
http://caniuse.com/#search=woff

We target IE 11+, so we don't need .svg, .ttf, .eoff

**Comment by vic-topcoder** [comment]
There are warnings listed in dev tools in Chrome/Firefox, so you can trace the problem. Since I don't know who can help, I added you all here. @vikasrohit can provide deeper feedback on the issue, but if we can't get our font stack sorted soon, we cannot render the UI effectively on people that don't have Roboto installed locally.

**Comment by vic-topcoder** [comment]
@vikasrohit: is there a reason we're still using Typekit on topcoder-app?

**Comment by birdofpreyru** [comment]
@vic-topcoder Ok, I'll fix it tomorrow.

By the way, have you checked our new `community-app`? Roboto fonts are already in there, together with the test page, which renders them all. In case you have not seen it yet, I'd say you should have a look and tell whether you like how the fonts are rendered there, or whether this need further improvements.

**Comment by vic-topcoder** [comment]
@birdofpreyru I'll take a look - haven't had time yet. That sounds amazing! Thanks!

**Comment by vic-topcoder** [comment]
I love the console and how it lists all resources generated. Yes, if it is possible to use this solution on all projects, that'll be great. Can you also post a link to the solution so we know how to keep all implementations in sync (for future reference)? Thanks!

**Comment by birdofpreyru** [comment]
Fixed in 252ce5a6b7134451e905c3b5c8a1d55b05b5f575
I've also added font test page into this repo at the `/font-test` route, so that we can easily verify that setup works in different environments.

> if it is possible to use this solution on all projects, that'll be great.

I guess, we should release `community-app` (with Topcoder specific staff excluded, but with more tests, examples and docs added) as starting pack for ReactJS and re-use it in all future projects. Embeding it into existing projects is tricky, as in the on-going TC projects webpack config and different assets are spread among various repos, so dependencies are messy, and this is the main source of problems.

In this issue with fonts, the webpack launched inside `topcoder-app` just didn't pack the font assets from `react-components`, and didn't complain about it. Also, the `webpack-dev-server` (`community-app` uses ExpressJS server with Morgan logger, which prints the logs you liked) which is used to serve the page in `topcoder-app` does not set 404 status when it server 404 error page, thus in dev environment the consol did show obscure warnings, rather than clear 404 error messages.
